src/modules/flow/console/console.c

```diff
@@ -159,7 +162,36 @@ console_in_process(struct sol_flow_node *node, void *data, uint16_t port, uint16
     } else {
         sol_flow_send_error_packet(node, -EINVAL, "Unsupported packet=%p type=%p (%s)",
             packet, packet_type, packet_type->name);
-        return 0;
+        return -EINVAL;
+    }
+    return 0;
+}
+
+static int
+console_in_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    int r;
+    struct console_data *mdata = data;
+    const struct sol_flow_packet_type *packet_type = sol_flow_packet_get_type(packet);
+
+    if (sol_flow_packet_is_composed_type(packet_type)) {
+        uint16_t len, i;
+        struct sol_flow_packet **packets;
+
+        sol_flow_packet_get_composed_members_len(packet_type, &len);
+        packets = malloc(len * sizeof(struct sol_flow_packet *));
+        SOL_NULL_CHECK(packets, -ENOMEM);
+        sol_flow_packet_get(packet, packets);
+        console_output(mdata, true, "Composed packet {");
+        for (i = 0; i < len; i++) {
+            r = print_packet_content(packets[i], node, mdata, false);
+            SOL_INT_CHECK(r, < 0, r);
+        }
+        console_output(mdata, false, "} (%s)", packet_type->name);
```

Comment:
I didn't test, but it appears to print suffix on first `console_output` call, is it ok? Suffix shouldn't be printed after the '}'? Or am I missing something?